iothub_client/inc/iothub_client_core_common.h

```diff
@@ -348,31 +348,55 @@ extern "C"
         const char* deviceSasToken;
     } IOTHUB_CLIENT_DEVICE_CONFIG;
 
+    /** @brief Current version of @p IOTHUB_CLIENT_COMMAND_REQUEST structure.  */
+    #define IOTHUB_CLIENT_COMMAND_REQUEST_STRUCT_VERSION_1 1
+
+    /** @brief    This struct specifies parameters of an incoming command request */  
+    typedef struct IOTHUB_CLIENT_COMMAND_REQUEST_TAG
+    {
+        /** @brief   Version of the structure.  Currently the SDK sets this to IOTHUB_CLIENT_COMMAND_REQUEST_STRUCT_VERSION_1. */
+        int structVersion;
+        /** @brief   Name of the component associated with this request.  If this is targeting the root component, this will be NULL. */
+        const char* componentName;
+        /** @brief   Name of the command associated with this request. */
+        const char* commandName;
+        /** @brief    Raw payload of the request.  This is NOT guaranteed to be a \0 terminated string. */
+        const unsigned char* payload;
+        /** @brief    Number of bytes of @p payload. */
+        size_t payloadLength;
+        /** @brief    Payload type of @p payload, if and only if it was explicitly indicated by IoT Hub.  This currently 
+                      will ALWAYS be NULL.  Future versions of the SDK may set this value based on metadata received from IoT Hub. */
+        const char* payloadContentType;
```

Comment:
Nit: the idea of having this structure approach is that we can add values in the future without breaking people. Should we really have this if it isn't even supported right now (and might not)?

Reply:
You're too modest - this isn't a nit but something I need to do.